crates/ruff_python_formatter/src/comments/placement.rs

```diff
@@ -415,6 +422,69 @@ fn handle_end_of_line_comment_around_body<'a>(
     CommentPlacement::Default(comment)
 }
 
+fn is_clause_header(node: AnyNodeRef) -> bool {
+    matches!(
+        node.kind(),
+        NodeKind::StmtIf
+            | NodeKind::StmtFor
+            | NodeKind::StmtWhile
+            | NodeKind::StmtTry
+            | NodeKind::StmtFunctionDef
+            | NodeKind::StmtWith
+            | NodeKind::ElifElseClause
+    )
+}
+
+fn handle_suppression_comment_on_clause_header<'a>(
+    comment: DecoratedComment<'a>,
+    locator: &Locator,
+) -> CommentPlacement<'a> {
+    // Ensure the comment is an end-of-line comment
+    if !comment.line_position().is_end_of_line() {
+        return CommentPlacement::Default(comment);
+    }
+
+    // Extract the comment text
+    let comment_text = locator.slice(comment.range());
+
+    // Check if the comment is a suppression comment (e.g., `# fmt: skip`)
+    if SuppressionKind::from_comment(comment_text).is_none() {
+        return CommentPlacement::Default(comment);
+    }
+
+    // Get the preceding node (e.g., `StmtIf`, `StmtFor`, etc.)
+    let Some(preceding_node) = comment.preceding_node() else {
+        return CommentPlacement::Default(comment);
+    };
+
+    // Handle clause headers with their own nodes (if, while, for, etc.)
+    if is_clause_header(preceding_node) {
+        // Check if the comment is on the same line as the clause header
+        let header_line_start = locator.line_start(preceding_node.start());
+        let comment_line_start = locator.line_start(comment.start());
+
+        if header_line_start == comment_line_start {
+            // The comment is on the same line as the clause header, apply fmt: skip to the entire clause
+            return CommentPlacement::trailing(preceding_node, comment);
+        }
+    }
+
+    // Handle special cases like `else`, `except`, `finally` blocks that are not their own nodes
+    if let Some(parent) = comment.enclosing_parent() {
+        if is_clause_header(parent) {
+            // Apply fmt: skip if the comment is on the same line as the 'else', 'except', or 'finally'
+            let header_line_start = locator.line_start(parent.start());
+            let comment_line_start = locator.line_start(comment.start());
```

Comment:
I think this is an over-simplification. E.g. 

```python
def test(
	a, 
	b, 
): ... # fmt: skip

def test() -> List[
	str,
	int,
]: ... # fmt: skip

if (
	a and b and c and [
		1, 
		2,
): ... # fmt: skip 
```

There's `ClauseHeader` that has some fairly complicated logic to find the exact range of the header

https://github.com/astral-sh/ruff/blob/138e70bd5c01d61061247c43b1bbb33d0290a18f/crates/ruff_python_formatter/src/statement/clause.rs#L40-L70

maybe we can reuse some of the logic here